packages/core/src/agent/agent.test.ts

```diff
@@ -1693,6 +1693,127 @@ describe('agent', () => {
   });
 
   describe('agent tool handling', () => {
+    it('should handle tool name collisions caused by formatting', async () => {
```

Comment:
Hmm I feel like this case we should throw an error, this will be confusing for the agent